src/v/v8_engine/script_dispatcher.h

```diff
@@ -0,0 +1,253 @@
+    model::topic_namespace topic;
+    ss::sstring function_name;
+    iobuf code;
+    size_t max_heap_size_bytes{10000};
+    size_t run_timeout_ms{200};
+
+    script_compile_params(
+      model::topic_namespace topic, ss::sstring function_name, iobuf code)
+      : topic(std::move(topic))
+      , function_name(std::move(function_name))
+      , code(std::move(code)) {}
+
+    script_compile_params(const script_compile_params& other)
+      : topic(other.topic)
+      , function_name(other.function_name)
+      , code(other.code.copy())
+      , max_heap_size_bytes(other.max_heap_size_bytes)
+      , run_timeout_ms(other.run_timeout_ms) {}
+};
+
+// This class implement database for v8_runtimes. It can be updated by
+// data_policy_manager. Fetch request will only run get for getting v8_runtimes
+template<typename Executor>
+class script_dispatcher {
+    constexpr static size_t _mutex_count{50};
+
+public:
+    explicit script_dispatcher(Executor& executor)
+      : _executor(executor) {}
+
+    ss::future<> start() {
+        (void)ss::with_gate(_gate, [this] {
+            return ss::do_until(
+              [this] { return _as.abort_requested(); },
+              [this]() -> ss::future<> { co_await update_runtimes(); });
+        });
+
+        co_return;
+    }
+
+    ss::future<> stop() {
+        _as.request_abort();
+        co_await _gate.close();
+    }
+
+    // Insert only put compilation params to map.
+    bool insert(
+      model::topic_namespace topic, ss::sstring function_name, iobuf code) {
+        return _compilation_params
+          .emplace(
+            topic,
+            script_compile_params(
+              topic, std::move(function_name), std::move(code)))
+          .second;
+    }
+
+    // Will delete information about dp from compilation_params map, Also delete
+    // runtime if it exists.
+    // If we change data_policy several times during compilation, we
+    // need to notify background workwe that his runtimes expired and background
+    // worker should not save this runtimes to map, because after that runtimes
+    // map will contain "old" v8.
+    bool remove(const model::topic_namespace& topic) {
+        bool res = _compilation_params.erase(topic) == 1;
+        if (!res) {
+            return res;
+        }
+
+        auto it = _runtimes.find(topic);
+        if (it == _runtimes.end()) {
+            return res;
+        }
+
+        ss::visit(
+          it->second,
+          [](runtimes_counter& counter) {
+              counter.is_compiled = false;
+              counter.deletion++;
+          },
+          [this, &it](const ss::lw_shared_ptr<script>&) {
+              _runtimes.erase(it);
+          });
+
+        return res;
+    }
+
+    /// This method return v8_runtimes for data-policy (aka topic on curent
+    /// moment). If task inside background queue it will return error.
+    result<ss::lw_shared_ptr<script>, errc>
+    get(const model::topic_namespace& topic) {
```

Comment:
I think we can just use promise/futures to return exactly when ready. Store a promise per call to the function and alert all callers interested for `topic` when the event is triggered

Reply:
Otherwise how will one know when it is ready? Busy waiting seems to be the only way.

Reply:
If fetch got v8_is_compiling in get method it should retry request. 
Also it does not block fetch session

Reply:
It will be cool
On current moment I an waiting Noah review. After that I think we can review it together and debate